tests/config/config_test.go

```diff
@@ -0,0 +1,117 @@
+package config
+
+import (
+	"context"
+	fmt "fmt"
+	"testing"
+	"time"
+
+	gorums "github.com/relab/gorums"
+	"google.golang.org/grpc"
+)
+
+type (
+	cfgSrv struct {
+		name string
+	}
+	cfgQSpec struct {
+		quorum int
+	}
+)
+
+func (srv cfgSrv) Config(ctx context.Context, req *Request, out func(*Response, error)) {
+	out(&Response{
+		Name: srv.name,
+		Num:  req.GetNum(),
+	}, nil)
+}
+
+func newQSpec(cfgSize int) *cfgQSpec {
+	return &cfgQSpec{quorum: cfgSize/2 + 1}
+}
+
+func (q cfgQSpec) ConfigQF(_ *Request, replies map[uint32]*Response) (*Response, bool) {
+	if len(replies) < q.quorum {
+		return nil, false
+	}
+	var reply *Response
+	for _, r := range replies {
+		reply = r
+		break
+	}
+	return reply, true
+}
+
+func setup(t *testing.T, mgr *Manager, cfgSize int) (cfg *Configuration, teardown func()) {
+	t.Helper()
+	srvs := make([]*cfgSrv, cfgSize)
+	for i := range srvs {
+		srvs[i] = &cfgSrv{}
+	}
+	addrs, closeServers := gorums.TestSetup(t, cfgSize, func(i int) gorums.ServerIface {
+		srv := gorums.NewServer()
+		RegisterConfigTestServer(srv, srvs[i])
+		return srv
+	})
+	for i := range srvs {
+		srvs[i].name = addrs[i]
+	}
+	cfg, err := mgr.NewConfiguration(newQSpec(cfgSize), gorums.WithNodeList(addrs))
+	if err != nil {
+		t.Fatal(err)
+	}
+	teardown = func() {
+		mgr.Close()
+		closeServers()
+	}
+	return cfg, teardown
+}
+
+func TestConfig(t *testing.T) {
```

Comment:
Some comments explaining the test would be helpful here I think. I looks like the `setup()` function is creating a new configuration with new nodes, such that each configuration created with `setup()` contain separate sets of nodes? Then you are testing the `With()` and `Without()` on these configurations?